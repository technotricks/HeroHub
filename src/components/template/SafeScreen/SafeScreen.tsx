import {StatusBar} from 'react-native';

import type {PropsWithChildren} from 'react';
import {MainContainer} from '@/components/atom';

function SafeScreen({children}: PropsWithChildren) {
  return (
    <MainContainer>
      <StatusBar barStyle={'light-content'} backgroundColor={'red'} />
      {children}
    </MainContainer>
  );
}

export default SafeScreen;
