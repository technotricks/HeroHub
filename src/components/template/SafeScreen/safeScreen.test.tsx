import 'react-native';
import React from 'react';

import {it, expect, describe} from '@jest/globals';

// Note: test renderer must be required after react-native.
import {act, create} from 'react-test-renderer';
import SafeScreen from './SafeScreen';

describe('SafeScreen component should render correctly', () => {
  const view = create(<SafeScreen></SafeScreen>);

  it('snapshot', () => {
    expect(view).toMatchSnapshot();
  });
});
