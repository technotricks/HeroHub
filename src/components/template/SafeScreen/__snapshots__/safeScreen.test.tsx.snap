// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafeScreen component should render correctly snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
/>
`;
